OpenRA.Mods.Common/Lint/CheckTranslationReference.cs

```diff
@@ -141,6 +142,23 @@ static TranslationKeys GetUsedTranslationKeysInRuleset(Ruleset rules)
 				}
 			}
 
+			foreach (var weapon in rules.Weapons)
+			{
+				foreach (var warhead in weapon.Value.Warheads)
+				{
+					var warheadType = warhead.GetType();
+					foreach (var field in Utility.GetFields(warheadType))
+					{
+						var translationReference = Utility.GetCustomAttributes<TranslationReferenceAttribute>(field, true).SingleOrDefault();
+						if (translationReference == null)
+							continue;
+
+						foreach (var key in LintExts.GetFieldValues(warhead, field, translationReference.DictionaryReference))
+							usedKeys.Add(key, translationReference, $"Weapon `{weapon.Key}` warhead {warheadType.Name[..^7]}.{field.Name}`");
```

Comment:
Missing a backtick at the startof the warhead name.

I'd also be tempted to leave out the `[..^7]`, just in case the name doesn't end in the `Warhead` suffix and it truncates oddly.

Reply:
That is some engine magic to always remove the `Warhead` suffix I think.

Reply:
Indeed, see https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Game/GameRules/WeaponInfo.cs#L165.